geometry-builder.cpp

```diff
@@ -760,8 +759,71 @@ void geometry_builder::set_exclude_broken_polygon(int exclude)
     excludepoly = exclude;
 }
 
+void geometry_builder::set_reprojection(struct reprojection *r)
+{
+    reprojection = r;
+}
+
+/**
+ * Computes area of given polygonal geometry.
+ * \return the area in projected units, or in EPSG 3857 if area reprojection is enabled
+ */
+double geometry_builder::getArea(const geos::geom::Geometry *geom) const
+{
+    // reprojection is not necessary, or has not been asked for.
+    if (!reprojection) {
+        return geom->getArea();
+    }
+
+    // MultiPolygon - return sum of individual areas
+    if (const geos::geom::MultiPolygon* multi = dynamic_cast<const geos::geom::MultiPolygon *>(geom)) {
```

Comment:
this is the multipolygon case, https://github.com/openstreetmap/osm2pgsql/pull/454/files#diff-853d1c6f0a63324f82abcb98373e8bc5 only contains a simple closed way.